components/shared/ModalComponent/ModalComponent.tsx

```diff
@@ -0,0 +1,22 @@
+import { useTranslation } from 'next-i18next';
+
+import { PropsType } from './types';
+import { SignUpForm, CheckEmail, AccountActivated } from 'components';
+
+const ModalComponent: React.FC<PropsType> = ({ component }) => {
+  const { t } = useTranslation();
+
+  if (component === 'signup') return <SignUpForm />;
```

Comment:
i legit can not understand what are u doing here. why can't u just pass <SignUpForm /> as componeent??